### PR TITLE
Introduce Service Instance resource

### PR DESCRIPTION
**Pull request by NickyMateev**
# Introduce Service Instance resource

## Motivation

Introduces Service Instance resource so Service Manager can start manage such entities.

## Pull Request status

* [x] Initial implementation

## Topic branch:
https://github.com/Peripli/service-manager/pull/351

### CODE REVIEW

File: pkg/types/service_instance.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+type ServiceInstance struct {
+	Base
+	Name            string          `json:"name"`
+	ServicePlanID   string          `json:"service_plan_id"`
+	PlatformID      string          `json:"platform_id"`
+	MaintenanceInfo json.RawMessage `json:"maintenance_info"`
+	Context         json.RawMessage `json:"-"`
+	PreviousValues  json.RawMessage `json:"-"`
+	Ready           bool            `json:"ready"`
+	Usable          bool            `json:"usable"`
```
Comment (dzahariev) [inline pkg/types/service_instance.go]: What is the semantic of this field

Comment (NickyMateev) [reply pkg/types/service_instance.go]: When a broker returns an error that the instance is no longer usable (due to an error during async update or deprovision) we also store that information in SM.

**Comment by NickyMateev** [comment]
@KirilKabakchiev 
How do you propose component testing GET/LIST when there are no Create/Update/Delete APIs yet? 
There's no direct way to create a resource, there's also no indirect way (eg. services and plans).

Also, the state of the controller right now is plain-vanilla base controller logic, nothing special about, which already has tests.

**Comment by KirilKabakchiev** [comment]
This is a new API and if you don't enable the list/get component tests its hard to guarantee that querying is working properly for all fields. I would suggest to implement the blueprint for instances by using the repository that is provided in the context

**Pull request merged**